enterprise/views.py

```diff
@@ -523,6 +550,29 @@ def get_final_price(self, mode, request):
             )
         return mode['original_price']
 
+    @staticmethod
+    def post_order_to_ecommerce(request, sku):
+        """
+        Post an order for audit enrollment  for the user.
+        """
+        try:
+            return post_audit_order_to_ecommerce(request.user, sku)
+        except (HTTPError, Timeout, HttpClientError):
+            messages.error(
+                request,
+                _(
+                    "There was an error completing your enrollment in the course, please try again. "
+                    "If the problem persists, contact {link_start}{platform_name} support{link_end}.",
+                ).format(
+                    link_start='<a href="{support_link}" target="_blank">'.format(
+                        support_link=configuration_helpers.get_value('ENTERPRISE_SUPPORT_URL')
+                    ),
+                    platform_name=configuration_helpers.get_value('PLATFORM_NAME'),
+                    link_end='</a>',
+                ),
+            )
+            raise
+
```

Comment:
Move the addition of the message to the message cache to the method on the ecommerce client mentioned above. No need to separate this out into two separate locations in this file.